### PR TITLE
Add Alma and Rocky Linux as supported host OSes

### PR DESCRIPTION
**Pull request by cybertron**
These two distributions are essentially what CentOS was pre-Stream,
so they should generally be compatible as well. They are documented
as "best effort" support though since I don't think we'll be running
them in CI.

### CODE REVIEW

**Comment by cybertron** [comment]
/hold

This got it past the validation stage, but something failed later on. Looks like I need to make further changes.

**Comment by andfasano** [comment]
Note that the default packet setup step creates an instance with Centos8.
If you want something else, you'll need at least a temporary test where you could set the [`PACKET_OS` variable](https://github.com/openshift/release/blob/cb80be4641325733af23c5b7320e92e068f1184d/ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-ref.yaml#L16) as required

**Comment by cybertron** [comment]
Okay, this one is working for me locally on Alma. Had to get a little hacky with the ansible python interpreter, but that should be a temporary thing until Ansible fixes it properly.

/hold cancel

File: 01_install_requirements.sh [hunk 1]
```diff
@@ -34,8 +34,10 @@ sudo dnf -y upgrade
 # shellcheck disable=SC1091
 source /etc/os-release
 export DISTRO="${ID}${VERSION_ID%.*}"
-if [[ $DISTRO == "centos8" ]]; then
-    echo "CentOS is not supported anymore. Please switch to CentOS Stream / RHEL / Rocky Linux"
+if [[ $DISTRO == "centos8" || $DISTRO == "almalinux8" || $DISTRO == "rocky8" ]]; then
+    if [[ $DISTRO == "centos8" ]]; then
```
Comment (elfosardo) [inline 01_install_requirements.sh:38]: let's move this before L37 ?

**Review by elfosardo — CHANGES_REQUESTED** [review]

File: 01_install_requirements.sh (hunk 1 above)
Comment (cybertron) [reply 01_install_requirements.sh]: Done

**Comment by cybertron** [comment]
/retest

Some of these jobs are no longer required, and I think all of the required ones have passed.

**Comment by cybertron** [comment]
/retest-required

Looks like issues getting a test environment.

**Comment by danwinship** [comment]
any reason this hasn't merged? CentOS 8 on Equinix no longer seems to work for me:

```
[root@dwinship ~]# dnf -y install git jq make
CentOS Linux 8 - AppStream                                            1.5 kB/s |  38  B     00:00    
Error: Failed to download metadata for repo 'appstream': Cannot prepare internal mirrorlist: No URLs in mirrorlist
```

**Comment by danwinship** [comment]
hm... nor does Rocky even with this PR:

```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_ansible.posix.selinux_payload_btu2jecb/ansible_ansible.posix.selinux_payload.zip/ansible_collections/ansible/posix/plugins/modules/selinux.py", line 100, in <module>
ModuleNotFoundError: No module named 'selinux'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "configfile": "/etc/selinux/config",
            "policy": "targeted",
            "state": "permissive",
            "update_kernel_param": false
        }
    },
    "msg": "Failed to import the required Python library (libselinux-python) on dwinship's Python /bin/python3.9. Please read the module documentation and install it in the appropriate location. If the required library is installed, but Ansible is using the wrong Python interpreter, please consult the documentation on ansible_python_interpreter"
}
```

`python3-libselinux` is installed though...

**Comment by cybertron** [comment]
Ah, that sounds like the same problem I ran into with Alma. Maybe the Python interpreter workaround is needed for Rocky too. I thought someone had already fixed that for Rocky but maybe not. I need to rebase this anyway, so I'll add Rocky to the conditional so you can try it.

**Comment by danwinship** [comment]
> I'll add Rocky to the conditional so you can try it.

I retried using Alma and it worked, and I managed to test what I was testing. I do need to go back and test an updated patch at some point but I'm not sure when I'm going to get to that...

**Comment by danwinship** [comment]
/lgtm
works with Rocky now too

**Comment by cybertron** [comment]
/retest-required

Infra failure

**Comment by cybertron** [comment]
Rebased to account for some recent changes in OS setup.

**Pull request merged**